volatility3/framework/plugins/linux/pidhashtable.py

```diff
@@ -44,17 +44,16 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
             ),
         ]
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.vmlinux = None
-        self.vmlinux_layer = None
-
     def _is_valid_task(self, task) -> bool:
-        return bool(task and task.pid > 0 and self.vmlinux_layer.is_valid(task.parent))
+        vmlinux = self.context.modules[self.config["kernel"]]
+        vmlinux_layer = self.context.layers[vmlinux.layer_name]
+        return bool(task and task.pid > 0 and vmlinux_layer.is_valid(task.parent))
```

Comment:
Why not just use the task's layer rather than the vmlinux layer (or has the task been made in a physical or process layer)?  `task.vol.layer_name` should get you back to the layer the task was created on?